pyresample/geometry.py

```diff
@@ -717,37 +717,282 @@ def _do_transform(src, dst, lons, lats, alt):
+        if y > 0:
+            list_side0 = (src_lons[1, :], src_lats[1, :], src_lons[0, :], src_lats[0, :])
+            list_side2 = (src_lons[-2, :], src_lats[-2, :], src_lons[-1, :], src_lats[-1, :])
+            extended_side0_lonlats = _get_extended_lonlats(*list_side0, npts=y)
+            extended_side2_lonlats = _get_extended_lonlats(*list_side2, npts=y)
+            dst_lats = np.concatenate((extended_side0_lonlats[1][::-1, :], dst_lats), axis=0)
+            dst_lats = np.concatenate((dst_lats, extended_side2_lonlats[1]), axis=0)
+            dst_lons = np.concatenate((extended_side0_lonlats[0][::-1, :], dst_lons), axis=0)
+            dst_lons = np.concatenate((dst_lons, extended_side2_lonlats[0]), axis=0)
+
+        # Extend on x direction (side1 and side3)
+        if x > 0:
+            list_side1 = (dst_lons[:, -2], dst_lats[:, -2], dst_lons[:, -1], dst_lats[:, -1])
+            list_side3 = (dst_lons[:, 1], dst_lats[:, 1], dst_lons[:, 0], dst_lats[:, 0])
+            extended_side1_lonlats = _get_extended_lonlats(*list_side1, npts=x, transpose=False)
+            extended_side3_lonlats = _get_extended_lonlats(*list_side3, npts=x, transpose=False)
+            dst_lats = np.concatenate((dst_lats, extended_side1_lonlats[1]), axis=1)
+            dst_lats = np.concatenate((extended_side3_lonlats[1][:, ::-1], dst_lats), axis=1)
+            dst_lons = np.concatenate((dst_lons, extended_side1_lonlats[0]), axis=1)
+            dst_lons = np.concatenate((extended_side3_lonlats[0][:, ::-1], dst_lons), axis=1)
+
+        # Back-conversion array as input format
+        lons, _ = _convert_2D_array(dst_lons, to=src_lons_format, dims=['y', 'x'])
+        lats, _ = _convert_2D_array(dst_lats, to=src_lats_format, dims=['y', 'x'])
+
+        # Add additional info if the source array is a DataArray
+        if isinstance(self.lats, xr.DataArray) and isinstance(self.lons, xr.DataArray):
+            lats.attrs = self.lats.attrs.copy()
+            lons.attrs = self.lons.attrs.copy()
+
+        # Return the extended SwathDefinition
+        return SwathDefinition(lons, lats)
+
+    def reduce(self, x=0, y=0):
+        """Reduce the swath definition along x (along-track) and y (across-track) dimensions.
+
+        By default, it does not reduce on any direction.
+        To reduce of n pixel on both sides of the across-track direction, call swath_def.reduce(x=0, y=2).
+        """
+        # Check input validity (ensure reduced area is at least 2x2)
+        height = self.lats.shape[0]
+        width = self.lats.shape[1]
+        x = int(x)
+        y = int(y)
+        if x < 0 or y < 0:
+            raise ValueError('x and y arguments must be positive integers.')
+        if x >= np.floor(width / 2):
+            max_x = int(np.floor(width / 2)) - 1
+            raise ValueError("""You can at maximum reduce the along-track direction (x)
+                             of SwathDef by {} pixels on each side.""".format(max_x))
```

Comment:
Correct me if I'm wrong, but doesn't using triple quotes retain the whitespace you used for indentation? So the message would end up having a ton of whitespace in the middle when it is reported to the user.

Reply:
Solved